app/mapping.py

```diff
@@ -62,14 +62,19 @@ def __init__(self, mapping_definition, mapping_type):
 
 
 def get_mapping(index_name, document_type):
-    # es.indices.get_mapping has a key for the index name, regardless of any alias we may be going via, so rather than
-    # use index_name, we access the one and only value in the dictionary using next(iter).
-    mapping_data = next(iter(es.indices.get_mapping(index=index_name, doc_type=document_type).values()))
-
-    # In ES <=5, there may be multiple mapping types per document type (kind-of) - this is confusing, but going away,
-    # see <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/removal-of-types.html>. For us, document_type
-    # === mapping_type, and there will be only one per index.
-    return Mapping(mapping_data, mapping_type=document_type)
+    try:
+        # es.indices.get_mapping has a key for the index name, regardless of any alias we may be going via, so rather
+        # than use index_name, we access the one and only value in the dictionary using next(iter).
+        mapping_data = next(iter(es.indices.get_mapping(index=index_name, doc_type=document_type).values()))
```

Comment:
I know this isn't the point of the PR, but this `next(iter(` thing feels weird. It would probably be clearer to just try to get the first element. We're handling a StopIteration error but we aren't really ever trying to iterate; we're just trying to get the first thing from a sequence.

Reply:
Get some python3, man. `'dict_values' object does not support indexing`.

Reply:
Ugh. Right.